src/routes/user/user.router.ts

```diff
@@ -0,0 +1,91 @@
+import * as express from 'express';
+
+import { HTTP_CODES } from '../../constants';
+import { IUser } from '../../models';
+import { validatePartialUser, validateUser } from '../../validations/validations';
+import { createUser, updateUser } from './helpers';
+
+export const userRouter = express.Router();
+
+const users = new Map<string, IUser>();
+
+const testUser: IUser = {
+    id: 'b5c8aaf5-2df7-4dfb-a44f-7c8e7b8da959',
+    login: 'Userlogin',
+    password: 'Xvfasdfvsdfv4',
+    age: 25,
+    isDeleted: false,
+};
+
+users.set(testUser.id, testUser);
+
+userRouter.get('/:id', (req, res) => {
+    const userId = req.params.id;
+
+    const user = users.get(userId);
+
+    if (user && !user.isDeleted) {
+        res.status(HTTP_CODES.OK).json(user);
+    } else {
+        res.status(HTTP_CODES.NOT_FOUND).send(`User with id: ${userId} not found`);
+    }
+});
+
+userRouter.post('/', validateUser, (req, res) => {
```

Comment:
I wouldn't expect to specify `isDeleted` field when creating a user
```sh
$ curl -X POST -H 'content-type: application/json' -d '{"login":"aaa","age":52}' 'http://localhost:3000/user'
{"validationErrors":[{"keyword":"required","dataPath":"","schemaPath":"#/definitions/UserToValidate/required","params":{"missingProperty":"isDeleted"},"message":"should have required property 'isDeleted'"}]}% 
```

Reply:
I expect to be able to create a user after deleting it
```sh
$ curl -X POST -H 'content-type: application/json' -d '{"login":"aaaaa","age":52,"isDeleted":false,"password":"2aAaaaaaaaaaaa"}' 'http://localhost:3000/user'
{"id":"bb79a794-be20-4eec-b491-ba15fa436a5e","login":"aaaaa","age":52,"isDeleted":false,"password":"2aAaaaaaaaaaaa"}
$ curl -X DELETE 'http://localhost:3000/user/bb79a794-be20-4eec-b491-ba15fa436a5e'
User with id: bb79a794-be20-4eec-b491-ba15fa436a5e is deleted
$ curl -X POST -H 'content-type: application/json' -d '{"login":"aaaaa","age":52,"isDeleted":false,"password":"2aAaaaaaaaaaaa"}' 'http://localhost:3000/user'
User with login aaaaa already exists% 
```

Reply:
@KonanMentor fixed